### DISCUSSION TITLE
Select View for any view.

### DISCUSSION THREAD

**Discussion by DrRuhe** [gyscos/cursive#631] [Q&A]
Hi,
is there a Select view for arbitrary Views?
So basically a combination between the ListView (ability to do arbitrary Views) and SelectViews (do selections and attach values to the items in the list).

I know there is the Select view, but that can only Display Styled Strings.
Kind Regards.

**Comment by gyscos** [comment]
Hi!

There is not currently such a view builtin.

Do you have an idea what it would be to use such a view? (Like a hypothetical example code using this view, not the implementation itself.)

**Reply by DrRuhe** [reply]
Hi, 
since this request is already a bit old I have locally tried to write such a view. As a base implementation, I used the current select_view.rs and replaced the StyledString in the Item with a Box<dyn View>. Then I started to fix all the errors that popped up and got to a compiling version. Sadly my limited understanding of the cursive internals meant that no items will ever be displayed (couldn't figure out the draw Method of the View trait). 

As to how the usage works, basically identical to the current Select View, just with the difference that I removed some string-specific constructors, and replaced them with methods that use Data that implements IntoBoxedView to populate both the Data and the representation.  

For the Use Case:
[![asciicast](https://asciinema.org/a/X9h8HKEModPJ3Dr8iBNIJ8JBg.svg)](https://asciinema.org/a/X9h8HKEModPJ3Dr8iBNIJ8JBg)
Currently I use a ListView for that. However, I want to be able to select things from the list. Due to the 3 columns it is not possible to use Styled Strings, especially, if the description in the 3rd column wraps around, which pushed the subsequent elements further down.


It would be great if we could find a way to collaborate on this, since I think this would greatly benefit cursive.
